types/wordpress__edit-post/package.json

```diff
@@ -8,9 +8,9 @@
     "minimumTypeScriptVersion": "5.0",
     "dependencies": {
         "@types/react": "*",
-        "@types/wordpress__editor": "*",
         "@wordpress/components": "^28.4.0",
         "@wordpress/data": "^10.4.0",
+        "@wordpress/editor": "^14.13.0",
         "@wordpress/element": "^6.4.0"
```

Comment:
FWIW these all should be `"*"`, and I think that's true for all of the other DT packages in the `@wordpress` scope; the upstream packages say `"*"` so DT should say `"*"`.

Not required for this PR but I was planning on sending that change as a followup.